lib/headings.js

```diff
@@ -44,15 +49,21 @@ var headings = module.exports = function (md, options) {
             .toLowerCase() // because `slug` doesn't lowercase
         )
 
-        // add some heading attributes
-        tokens[idx].attrSet('id', headings.prefix + postfix)
-        tokens[idx].attrJoin('class', headings.className)
+        // add 3 new token objects link_open, text, link_close
+        var link_open = new Token('link_open', 'a', 1)
+        var text = new Token('text', '', 0)
+        text.content = "link"
+        var link_close = new Token('link_close', 'a', -1)
+        
+        // add some link attributes
+        link_open.attrSet('id', headings.prefix + postfix)
+        link_open.attrSet('class', headings.className)
+        link_open.attrSet('href', "#" + postfix)
 
-        // if the heading doesn't already contain a link, wrap the contents in one
-        if (children.filter(isHtmlLink).length === 0) {
-          tokenUtil.wrap(children, postfix)
```

Comment:
also, so I don't forget, we'll be able to remove the `wrap` function from token-util.js now too.

delete all teh codez